libr/io/io.c

```diff
@@ -889,25 +794,7 @@ R_API bool r_io_is_blockdevice(RIO *io) {
 }
 
 R_API ut64 r_io_size(RIO *io) {
-	int oldva;
-	ut64 size, here;
-	if (!io) return 0LL;
-	oldva = io->va;
-	if (r_io_is_listener (io)) {
-		return UT64_MAX;
-	}
-	io->va = false;
-	here = r_io_seek (io, 0, R_IO_SEEK_CUR);
-	size = r_io_seek (io, 0, R_IO_SEEK_END);
-	if (r_io_seek (io, here, R_IO_SEEK_SET) != here) {
-		eprintf ("Failed to reset the file position\n");
-	}
-	io->va = oldva;
-	if (r_io_is_blockdevice (io)) {
```

Comment:
Without this opening a block device will fail because it will consider the size of 0 and not read beyond